controllers/rcj_soccer_referee_supervisor/referee/referee.py

```diff
@@ -1,25 +1,32 @@
 from referee.consts import (
     GOAL_X_LIMIT,
     TIME_STEP,
-    ROBOT_NAMES
+    ROBOT_NAMES,
 )
 from referee.supervisor import RCJSoccerSupervisor
 
 
 class RCJSoccerReferee(RCJSoccerSupervisor):
     def check_robots_in_penalty_area(self):
-        """Check whether robots are violating rule not to stay in
-        penalty area for longer period of time"""
-        # TODO: decide if this rule will be applied and then implement it
-        pass
+        """
+        Check whether robots are violating rule not to stay in
+        penalty area for longer period of time
+        """
+        for robot in ROBOT_NAMES:
+            pos = self.robot_translation[robot]
+            self.penalty_area_chck[robot].track(pos, self.time)
+
+            if self.penalty_area_chck[robot].is_violating():
+                print(f'Robot {robot}: Inside penalty area for too long')
+                # TODO: move the robot to the FURTHEST unoccupied neutral spot
+                self.reset_robot_position(robot)
```

Comment:
We will need to find *furthest* unoccupied neutral spot here.

Reply:
@Adman I guess a prerequisite would be to find out what's the situation like with all the neutral spots (i.e. are their occupied or not). That sounds like a separate module -- the results will probably be useful for other "rules" as well.

What do you think? Worth another PR or will we do it here?

Reply:
@mrshu I would prefer separate PR, where we could add functionality for finding the closest neutral spot.

Reply:
@Adman makes sense, let's get this merged then!